sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -108,6 +123,11 @@ private async ValueTask<bool> AuthorizeRequestOnChallengeAsyncInternal(HttpMessa
                 {
                     return false;
                 }
+                else if (claims is not null)
+                {
+                    _challenge.Claims = claims;
+                    s_challengeCache[authority] = _challenge;
```

Comment:
The policy should use claims only once, in the token request following the 401 response that provided them. They don't apply to other token requests and aren't cacheable

Reply:
Got it, I'm going to remove this condition then.